### PR TITLE
fix: Extract screen sharing state to callState (SQCALL-353)

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

**Review by ffflorian — APPROVED** [review]

File: src/script/components/calling/CallingOverlayContainer.tsx [hunk 1]
```diff
@@ -132,18 +131,18 @@ const CallingContainer: React.FC<CallingContainerProps> = ({
     }
     const showScreenSelection = (): Promise<void> => {
       return new Promise(resolve => {
-        onChooseScreen = (deviceId: string): void => {
+        callState.onChooseScreen = (deviceId: string): void => {
```
Comment (Yserz) [inline src/script/components/calling/CallingOverlayContainer.tsx:134]: Mh, I think `onChooseScreen` shouldn't be in the callSate.

File: src/script/calling/CallState.ts [hunk 2]
```diff
@@ -31,6 +32,10 @@ export class CallState {
   public readonly acceptedVersionWarnings: ko.ObservableArray<string>;
   public readonly cbrEncoding: ko.Observable<number>;
   public readonly videoSpeakersActiveTab: ko.Observable<string>;
+  readonly selectableScreens: ko.Observable<ElectronDesktopCapturerSource[]>;
+  readonly selectableWindows: ko.Observable<ElectronDesktopCapturerSource[]>;
+  readonly isChoosingScreen: ko.PureComputed<boolean>;
```
Comment (Yserz) [inline src/script/calling/CallState.ts:37]: This looks more like a `canChooseScreen`, `hasScreensToChooseFrom` or `hasMultipleScreens`.

File: src/script/calling/CallState.ts [hunk 3]
```diff
@@ -39,12 +44,18 @@ export class CallState {
...
     this.acceptedVersionWarnings = ko.observableArray<string>();
-
     this.activeCalls.subscribe(activeCalls => {
       const activeCallIds = activeCalls.map(call => call.conversationId);
       this.acceptedVersionWarnings.remove(acceptedId => !activeCallIds.includes(acceptedId));
     });
     this.cbrEncoding = ko.observable(0);
     this.videoSpeakersActiveTab = ko.observable(VideoSpeakersTab.ALL);
+
+    this.onChooseScreen = (deviceId: string) => {};
```
Comment (Yserz) [inline src/script/calling/CallState.ts]: Mh, I think `onChooseScreen` shouldn't be in the callSate. It's kind of an "action" and should live in the repository. The callState should just hold the resulting state (e.g. currently selected deviceId).

**Review by Yserz — CHANGES_REQUESTED** [review]

File: src/script/calling/CallState.ts (hunk 3 above)
Comment (thisisamir98) [reply src/script/calling/CallState.ts]: https://github.com/wireapp/wire-webapp/pull/11438/commits/e5637e40cc55feefc3936b76ac2a8f32bc65d19f

**Review by thisisamir98 — COMMENTED** [review]
thanks for reviews @Yserz

**Pull request merged**